lib/util/itinerary.js

```diff
@@ -228,11 +228,24 @@ export function getItineraryBounds (itinerary) {
   return latLngBounds(coords)
 }
 
+/**
+ * Return a leaflet LatLngBounds object that encloses the given leg's geometry.
+ */
 export function getLegBounds (leg) {
-  return latLngBounds(polyline
+  let coords = polyline
     .toGeoJSON(leg.legGeometry.points)
     .coordinates.map(c => [c[1], c[0]])
-  )
+
+  // in certain cases, there might be zero-length coordinates in the leg
+  // geometry. In these cases, build us an array of coordinates using the from
+  // and to data of the leg.
+  if (coords.length === 0) {
+    coords = [
+      [leg.from.lat, leg.from.lon],
+      [leg.to.lat, leg.to.lon]
+    ]
+  }
```

Comment:
This is a bit of a nit picky style thing, but I kind of like the idea of declaring `coords` with `const` and if the length is zero, pushing the leg endpoints into the array:

```
coords.push([leg.from.lat, leg.from.lon], [leg.to.lat, leg.to.lon])
```

Reply:
huh, I never knew you could push one _or more_ elements using the `push` method.